### PR TITLE
Resize terminal

### PR DESCRIPTION
**Pull request by barrymcgee**
## Done

Added ability to drag to resize terminal on model details pages

## QA

- Pull code
- Run `./run clean && ./run serve`
- Open http://0.0.0.0:8036/
- Go to model details page and drag the top of terminal window

### CODE REVIEW

**Comment by anthonydillon** [comment]
@barrymcgee pushed a fix remove the event listener on mouseup. The issue was the listeners did not match. The way js knows it's the same is the combination of the target and the listener function signature.

File: src/components/Terminal/Terminal.js [hunk 1]
```diff
@@ -23,40 +31,79 @@ const Terminal = ({ address, modelName }) => {
...
+        document.addEventListener("mousemove", resize);
+      }
+    };
+
+    const removeResizeListener = () => {
+      document.removeEventListener("mousemove", resize);
+    };
+
+    // set resize listeners
+    window.addEventListener("mousedown", addResizeListener);
```
Comment (hatched) [inline src/components/Terminal/Terminal.js]: Why are these events not registered using the react event system? That way it should handle attaching and detaching and simplify this event handling code.

```
<div className="p-terminal__header" onMouseUp={...} onMouseDown={...}>
```

File: src/components/Terminal/Terminal.js [hunk 2]
```diff
@@ -23,40 +31,79 @@ const Terminal = ({ address, modelName }) => {
...
+    };
     // Because the user can switch the model details UI from within the
     // component the `modelName` passed above will change as the UI
     // updates. This causes the terminal to log in again which is
     // undesirable.
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
-  const [terminalVisible, setTerminalVisible] = useState(false);
+  useEffect(() => {
```
Comment (hatched) [inline src/components/Terminal/Terminal.js]: Can you add a comment that this `useEffect` order is important?

**Comment by hatched** [comment]
QA Notes:

When you click on the handle to drag it, it jumps to the bottom of the header. It would be nice if we could enhance the logic here so that it would drag from where the user clicks instead of jumping. Can you create a follow-up issue for this please.

File: src/components/Terminal/Terminal.js (hunk 1 above)
Comment (barrymcgee) [reply src/components/Terminal/Terminal.js]: Issue raised: https://github.com/canonical-web-and-design/jaas-dashboard/issues/397

**Pull request merged**